internal/plugin/platform.go

```diff
@@ -428,17 +429,37 @@ func (s *platformServer) Deploy(
 	internal := s.internal()
 	defer internal.Cleanup.Close()
 
+	// NOTE(izaak): Argmapper has two different ways to get a DeclaredResourcesResp:
+	// the value we provide below, or converting the pb.DeclaredResources supplied by core
+	// with a mapper function. In practice, it uses this one (which is important), but I don't
+	// know if we can rely on that behavior. The commented-out snippet below removes the other arg.
+	declaredResourcesResp := &component.DeclaredResourcesResp{}
+
+	// Remove the DeclaredResources arg from the list - we don't want to use that one.
+	//var realArgs []*proto.FuncSpec_Value
+	//for _, arg := range args.Args {
+	//	if arg.Type == "hashicorp.waypoint.sdk.DeclaredResources" {
+	//		continue
+	//	}
+	//	realArgs = append(realArgs, arg)
+	//}
+
 	encoded, raw, err := callDynamicFuncAny2(s.Impl.DeployFunc(), args.Args,
 		argmapper.ConverterFunc(s.Mappers...),
 		argmapper.Typed(internal),
 		argmapper.Typed(ctx),
+		argmapper.Typed(declaredResourcesResp),
 	)
 	if err != nil {
 		return nil, err
 	}
 
 	result := &proto.Deploy_Resp{Result: encoded, Deployment: &proto.Deploy{}}
 
+	result.DeclaredResources = &proto.DeclaredResources{
+		Resources: declaredResourcesResp.DeclaredResources,
```

Comment:
So at this point, this is still an empty `DeclaredResourcesResp{}`, right? Is there a reason we can't set it on line 457 with the other parts of the `Deploy_Resp`?

Reply:
It has been populated at this point! But we could set it on line 457 with the other fields as we make the result.